File: search/recursive-search.js
/**
 * recursive search algorith
 * recursive search works  for sorted items in a list
 * recusrive function takes parameters; array, begin,end and search item
 */

const array = [16, 19, 20, 23, 45, 56, 78, 90, 96, 100];
function recursiveSearch(array, begin,end,item) {
  var mid =  Math.floor((end+begin)/2);
  if(end >= begin) {
      if(array[mid] == item) {
        return mid;
      }
      else if(array[mid] > item) {
        return recursiveSearch(array,begin, mid-1,item);
      }
      else {
        return recursiveSearch(array,mid+1,end,item);
      }
  }
}
var flag = recursiveSearch(array,0,(array.length-1),100);
if(flag>=0) {
  console.log("match found at position %d\n",flag+1);
}
else {
  console.log("No match found..\n");
}

File: search/recursive-search.c
#include <stdio.h>
#include "string.h"
#include "stdlib.h"

int recursive(int *arr,int begin, int end, int item);
int main(void) {
  int arr[10] = {16, 19, 20, 23, 45, 56, 78, 90, 96, 100};
  int flag = -1;
  int begin = 0, end = (sizeof(arr)/sizeof(int))-1,item = 50;
   flag = recursive(arr,begin,end,item);
   if(flag>=0) {
     printf("Match found at position %d\n",flag+1);
     return 0;
   }
   printf("No match found..\n");

  return 0;
}
int recursive(int *arr,int begin, int end, int item){
  int mid = (end+begin)/2;
  if(end>=begin){
      if(arr[mid] == item) {
        return mid;
      }
      else if(arr[mid] > item) {
        return recursive(arr,begin,mid-1,item);
      }
      else {
        return recursive(arr,mid+1,  end, item);
      }
  }
  printf("end cannnot be less than beginning\n");
  return -1;
}